lib/validateSchema.js

```diff
@@ -30,22 +32,20 @@ function validateSchema(schema, options) {
 }
 
 function validateObject(schema, options) {
-	var validate = ajv.compile(schema);
-	var valid = validate(options);
+	const validate = ajv.compile(schema);
+	const valid = validate(options);
 	return valid ? [] : filterErrors(validate.errors);
 }
 
 function filterErrors(errors) {
-	var newErrors = [];
-	errors.forEach(function(err) {
-		var dataPath = err.dataPath;
-		var children = [];
-		newErrors = newErrors.filter(function(oldError) {
-			if(oldError.dataPath.indexOf(dataPath) >= 0) {
+	let newErrors = [];
+	errors.forEach((err) => {
+		const dataPath = err.dataPath;
+		let children = [];
+		newErrors = newErrors.filter((oldError) => {
+			if(oldError.dataPath.includes(dataPath)) {
 				if(oldError.children) {
-					oldError.children.forEach(function(child) {
-						children.push(child);
-					});
+					children = children.concat(oldError.children.slice(0));
```

Comment:
there was previously a comment to make this line `children = oldError.children.slice(0);`, but I think that would only give the last iteration's errors. It looks like children is an accumulator across many children, so I just made it concat with the recommended way.